xxhash.h

```diff
@@ -2679,7 +2679,11 @@ XXH_PUBLIC_API XXH64_hash_t XXH64_hashFromCanonical(const XXH64_canonical_t* src
 
 /* ===   Compiler specifics   === */
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   /* >= C99 */
+#if ((defined(sun) || defined(__sun)) && __cplusplus) /* Solaris includes __STDC_VERSION__ with C++. Tested with GCC 5.5 */
+#  define XXH_RESTRICT /* disable */
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   /* >= C99 */
+#  define XXH_RESTRICT /* disable */
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   /* >= C99 */
```

Comment:
Is this right? L2684 and L2686 look like they're testing the same condition.

Reply:
Good point @nealrichardson !
This is actually incorrect.
The first test shadows the second one
resulting in `XXH_RESTRICT` being always disabled.

Reply:
IIUC you don't need to revert the whole thing, you just want to prune L2685-L2686 right?

Reply:
Yes. The same PR, but with these 2 lines removed.

Reply:
👍 I can do that; in a strange coincidence I happened upon this exact issue this weekend myself.

Reply:
My bad! Sorry that's embarassing!